core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -540,21 +546,17 @@ def stack_new_level(
             **core_kwargs,
         )
 
-        if X_val is None:
-            aux_models = self.stack_new_level_aux(
-                X=X, y=y, base_model_names=core_models, level=level + 1, infer_limit=infer_limit, infer_limit_batch_size=infer_limit_batch_size, **aux_kwargs
-            )
-        else:
-            aux_models = self.stack_new_level_aux(
-                X=X_val,
-                y=y_val,
-                fit=False,
-                base_model_names=core_models,
-                level=level + 1,
-                infer_limit=infer_limit,
-                infer_limit_batch_size=infer_limit_batch_size,
-                **aux_kwargs,
-            )
+        aux_models = []
+        if full_weighted_ensemble:
+            full_aux_kwargs = aux_kwargs.copy()
+            if additional_full_weighted_ensemble:
```

Comment:
`and level > 1`

Reply:
I moved that check to [here](https://github.com/LennartPurucker/autogluon/blob/full_l3_post_hoc_ensemble/core/src/autogluon/core/trainer/abstract_trainer.py#L376) as I think it is the part of the code with more control and knowledge of the stack levels.

Reply:
Makes sense thanks